src/System.Runtime/src/System/HashCode.cs

```diff
@@ -0,0 +1,250 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Runtime;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    public unsafe struct HashCode
+    {
+        private const uint Queue = 4;
+        private const uint Length = 7;
+
+        private const uint Prime1 = 2654435761U;
+        private const uint Prime2 = 2246822519U;
+        private const uint Prime3 = 3266489917U;
+        private const uint Prime4 = 0668265263U;
+        private const uint Prime5 = 0374761393U;
+        
+        // 8x8 = 64 bytes = 1 x86 cache line
+        // +----+----+----+----+----+----+----+----+
+        // | 00 | 01 | 02 | 03 | 04 | 05 | 06 | 07 |
+        // | xxHash State      | Queue        | Le |
+        // +----+----+----+----+----+----+----+----+
+        private fixed uint _state[8];
+        
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [TargetedPatchingOptOut("Performance critical for inlining across NGen images.")]
+        private void Add(int value)
+        {
+            unchecked
+            {
+                var val = (uint)value;
+
+                fixed (uint* state = _state)
```

Comment:
As a convention we called such fixed pointers pXxx (pState in this example). It makes it easier to see in following code where there is a danger of overruns.